### ISSUE TITLE
cilium and kube-proxy keep restarting on a certain machine

### ISSUE THREAD

**Issue by swhoro** [cilium/cilium#27965]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

I have 4 machines in cluster, other 3 are working fine but one keep restarting cilium, cilium-operator and kube-proxy because of probe failed

### Cilium Version

cilium-cli: v0.15.3 compiled with go1.20.4 on linux/amd64
cilium image (default): v1.13.4
cilium image (stable): v1.14.1
cilium image (running): 1.14.1

### Kernel Version

all 4 machines:
6.1.0-11-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.38-4 (2023-08-08) x86_64 GNU/Linux

### Kubernetes Version

all 4 machines:
Client Version: v1.28.1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.1

### Sysdump

[cilium-sysdump-20230906-203901.zip](https://github.com/cilium/cilium/files/12537423/cilium-sysdump-20230906-203901.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
Hmm, well, two things:
- Cilium 1.14.1 actually only supports up to Kubernetes 1.27, and it seems some weird things can happen if you use 1.14.1 and Kubernetes 1.28. We've mainly been seeing issues with coredns though, not with probes failing.
- Is there anything else different about that node? What node is it so that whoever looks at the sysdump can focus in on the right one?

**Comment by swhoro** [comment]
> Hmm, well, two things:
> 
> * Cilium 1.14.1 actually only supports up to Kubernetes 1.27, and it seems some weird things can happen if you use 1.14.1 and Kubernetes 1.28. We've mainly been seeing issues with coredns though, not with probes failing.
> * Is there anything else different about that node? What node is it so that whoever looks at the sysdump can focus in on the right one?

Difference:
1.All of other nodes do not have gpu, this node has a 4090gpu, and this node use nvidia runtime, the others use default runc.
2.I have another gpu node joined in the cluster, it has the same hardware as the not working one, however it use default runc rather than nvidia runtime.
3.I once installed `metallb` and [ingress nginx controller](https://kubernetes.github.io/ingress-nginx/deploy/) using this not working machine(these two things cannot work properly and I do not know the reson). It seems before I installed these two things, the node was working properly, but after I installed, it keeps restarting these two pod(even after rejoining the cluster and rebuilding the cluster). I am not sure that it is the root cause.

**Comment by youngnick** [comment]
Oh, I didn't mention that Cilium 1.14.x is only tested with Kubernetes 1.27, not 1.28 - support for 1.28 has been merged into `main`, but is not present in 1.14 yet.

**Comment by aanm** [comment]
@swhoro would you be able to test it again with Kubernetes 1.28.2? Thank you. This seems related with #27982

**Comment by swhoro** [comment]
> @swhoro would you be able to test it again with Kubernetes 1.28.2? Thank you. This seems related with #27982

I have reinstalled the machine with brand new debian system. It seems the problem has been solved partially, the kube-proxy won't restart anymore and the cilium kept restarting a few days ago(restart about 200 times) but runs normally now.

What's more, I have updated all machines to k8s version 1.28.2 and cilium version 1.15.0-pre.0 before reinstalled the machine and it seems the problem won't be solved. Maybe this problem is caused by something I did to linux?
